modules/gateway/gateway_test.go

```diff
@@ -6,21 +6,27 @@ import (
 
 	"github.com/NebulousLabs/Sia/consensus"
 	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/modules/tester"
 )
 
-var port = 9001
+var (
+	rpcPort = 10000
+)
 
-func newTestGateway(t *testing.T) *Gateway {
-	g, err := New(":"+strconv.Itoa(port), consensus.CreateGenesisState(), "")
+// newTestingGateway returns a gateway read to use in a testing environment.
+func newTestingGateway(directory string, t *testing.T) *Gateway {
+	gDir := tester.TempDir(directory, modules.GatewayDir)
```

Comment:
I suppose you could take it a step further and use `runtime` to automatically get the name of the Test* function and use that for the folder name. Then you wouldn't need to pass it in manually. Probably overkill though.